### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by gerryhzgtest2021**

### CODE REVIEW

File: example.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module "vpc" {
+  source = "./modules/vpc"
+
+  az                 = ["us-east-1a", "us-east-1b"]
+  private_cidr_block = ["10.0.2.0/24", "10.0.3.0/24"]
+  private_number     = 2
+  public_cidr_block  = ["10.0.0.0/24", "10.0.1.0/24"]
+  public_number      = 2
+  vpc_cidr_block     = "10.0.0.0/16"
+}
```
Comment (vladimirmukhin) [inline example.tf:10]: Always finish files with a new line so there is no ⛔ signs in github diff

File: modules/vpc/main.tf [hunk 2]
```diff
@@ -20,25 +13,25 @@ resource "aws_vpc" "main" {
 
 #Subnets
 resource "aws_subnet" "main-public" {
-  count = 2
+  count = var.public_number
```
Comment (vladimirmukhin) [inline modules/vpc/main.tf:16]: use the following construction istead
```count = length(var.public_cidr_block)```

File: modules/vpc/main.tf [hunk 3]
```diff
@@ -20,25 +13,25 @@ resource "aws_vpc" "main" {
...
+  availability_zone       = var.az[count.index]
 
   tags = {
     Name = "main-public-${count.index + 1}"
   }
 }
 
 resource "aws_subnet" "main-private" {
-  count = 2
+  count = var.private_number
```
Comment (vladimirmukhin) [inline modules/vpc/main.tf:29]: same as my previous comment and for any other count in this module

File: modules/vpc/variables.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  description = "Range of the available zones"
+  type        = list(string)
+}
+
+variable "vpc_cidr_block" {
+  description = "The cidr block of vpc"
+  type        = string
+}
+
+variable "public_number" {
```
Comment (vladimirmukhin) [inline modules/vpc/variables.tf:23]: the "public_number" variable will not be needed after you fix my comments regarding count

File: modules/vpc/variables.tf [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  description = "the count number of public subnet"
+  type        = number
+}
+
+variable "private_number" {
```
Comment (vladimirmukhin) [inline modules/vpc/variables.tf:28]: same as public number

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: example.tf (hunk 1 above)
Comment (gerryhzgtest2021) [reply example.tf]: fixed

File: modules/vpc/main.tf (hunk 2 above)
Comment (gerryhzgtest2021) [reply modules/vpc/main.tf]: fixed

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.terraform.lock.hcl
+.terraform
+.idea
```
Comment (vladimirmukhin) [inline .gitignore:3]: Trailing newline should be added to .gitignore as well

File: modules/vpc/main.tf [hunk 7]
```diff
@@ -20,25 +13,25 @@ resource "aws_vpc" "main" {
...
 
   vpc_id                  = aws_vpc.main.id
-  cidr_block              = local.public_cidr_block[count.index]
+  cidr_block              = var.public_cidr_block[count.index]
   map_public_ip_on_launch = "true"
-  availability_zone       = local.az[count.index]
+  availability_zone       = var.az[count.index]
 
   tags = {
     Name = "main-public-${count.index + 1}"
```
Comment (vladimirmukhin) [inline modules/vpc/main.tf:24]:
```suggestion
    Name = "${var.env_code}-public-${count.index + 1}"
```

Comment (vladimirmukhin) [reply modules/vpc/main.tf:24]: Please do the same change the Name tag of each resource

Comment (vladimirmukhin) [reply modules/vpc/main.tf:24]: This is required for scalability. If the name is not parametrized we will end up with several VPCs called main when we deploy more than one VPC

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: modules/vpc/main.tf (hunk 3 above)
Comment (gerryhzgtest2021) [reply modules/vpc/main.tf]: fixed

File: modules/vpc/variables.tf (hunk 4 above)
Comment (gerryhzgtest2021) [reply modules/vpc/variables.tf]: fixed

File: modules/vpc/variables.tf (hunk 5 above)
Comment (gerryhzgtest2021) [reply modules/vpc/variables.tf]: fixed

File: .gitignore (hunk 6 above)
Comment (gerryhzgtest2021) [reply .gitignore]: fixed

File: modules/vpc/main.tf (hunk 7 above)
Comment (gerryhzgtest2021) [reply modules/vpc/main.tf]: fixed

**Pull request merged**